library/src/test/java/io/ona/kujaku/layers/ArrowLineLayerTest.java

```diff
@@ -413,7 +411,168 @@ public void sortFeaturesShouldSortDescByDateTime() throws InvalidArrowLineConfig
+                , ReflectionHelpers.ClassParameter.from(FeatureCollection.class, featureCollection)
+                , ReflectionHelpers.ClassParameter.from(ArrowLineLayer.SortConfig.class, sortConfig)
+        );
+
+        List<Feature> sortedFeatures = arrowHeadFeatureCollection.features();
+        assertEquals(5, sortedFeatures.size());
+
+        for (int i = 0; i < 5; i++) {
+            Feature sortedFeature = sortedFeatures.get(i);
+            assertEquals((int) sortedFeature.getNumberProperty("position"), i);
+        }
+    }
+
+    @Test
+    public void sortFeaturesShouldSortAscByNumber() throws InvalidArrowLineConfig {
+        ArrayList<Feature> featuresList = new ArrayList<>();
+
+        featuresList.add(generateRandomFeatureWithProperties(new GeoJSONFeature.Property("sample-number", 2000.67)
+                , new GeoJSONFeature.Property("position", 4)));
+        featuresList.add(generateRandomFeatureWithProperties(new GeoJSONFeature.Property("sample-number", 1900)
+                , new GeoJSONFeature.Property("position", 3)));
+        featuresList.add(generateRandomFeatureWithProperties(new GeoJSONFeature.Property("sample-number", 1200)
+                , new GeoJSONFeature.Property("position", 0)));
+        featuresList.add(generateRandomFeatureWithProperties(new GeoJSONFeature.Property("sample-number", 1800.989348394)
+                , new GeoJSONFeature.Property("position", 2)));
+        featuresList.add(generateRandomFeatureWithProperties(new GeoJSONFeature.Property("sample-number", 1500)
+                , new GeoJSONFeature.Property("position", 1)));
+
+        FeatureCollection featureCollection = FeatureCollection.fromFeatures(featuresList);
+
+        ArrowLineLayer.FeatureConfig featureConfig = new ArrowLineLayer.FeatureConfig(featureCollection);
+        ArrowLineLayer.SortConfig sortConfig = new ArrowLineLayer.SortConfig("sample-number"
+                , ArrowLineLayer.SortConfig.SortOrder.ASC
+                , ArrowLineLayer.SortConfig.PropertyType.NUMBER);
+
+        ArrowLineLayer.Builder builder = new ArrowLineLayer.Builder(context, featureConfig, sortConfig);
+        ArrowLineLayer arrowLineLayer = builder.build();
+
+        FeatureCollection arrowHeadFeatureCollection = ReflectionHelpers.callInstanceMethod(arrowLineLayer
+                , "sortFeatures"
+                , ReflectionHelpers.ClassParameter.from(FeatureCollection.class, featureCollection)
+                , ReflectionHelpers.ClassParameter.from(ArrowLineLayer.SortConfig.class, sortConfig)
+        );
+
+        List<Feature> sortedFeatures = arrowHeadFeatureCollection.features();
+        assertEquals(5, sortedFeatures.size());
+
+        for (int i = 0; i < 5; i++) {
+            Feature sortedFeature = sortedFeatures.get(i);
+            assertEquals((int) sortedFeature.getNumberProperty("position"), i);
+        }
+    }
+
+    @Test
+    public void sortFeaturesShouldSortAscByString() throws InvalidArrowLineConfig {
```

Comment:
Generally, I don't think a test should throw an exception unless you expect it to.

Why not add a try-catch block?

Reply:
Exactly, so why catch it if it's not supposed to be thrown for an passing test

Reply:
I think the point was that the test should have correct inputs that should ensure it does not fail because of an invariant that is not under test.

The unit test should not fail because of something that the unit test is not testing.  In this case, it should only fail if the the sort did not happen as expected, like maybe wrong ordering. Not because the input was wrong.

The try-catch is simply because the originating function throws one not because we expect the test to throw one.

Reply:
Right, the test should not fail. Why catch the exception or rather handle the exception if we don't expect it to fail? Do we want it to fail silently or continue passing even if an exception is thrown?

Reply:
It should not fail in the first place, I guess is what I'm trying to say i.e. if it is to fail, it should be for no other reason than the one under test.

The `try-catch` is not so as to handle the exception (the catch can be empty in this case) it's just because Java expects you to either wrap a method call that throws an exception in this block or in turn, throw the exception in the calling method i.e. it's just a formality to allow compiling.

The idea here would be that a developer is expected to know what are valid inputs for a method under test and provide those that wouldn't throw an exception where the exception is not under test.

The developer can then create a separate test to ensure that no exception is thrown for what they believe is valid input for a method under test. This method can have the `throws` signature if the developer chooses.

In short, let every test case test one thing only, and the method signature should clearly show what is under test. Adding a `throws exception` implies that an exception should be expected, when it shouldn't. Thoughts?

Reply:
Cool, so we add an empty `try-catch` and then if the developer breaks or changes the functionality, the test either passes(if we are helping it pass by catching the unexpected exception) or fails. If it fails, then it fails with no information on why it failed. The only information will be that it did not meet the assert condition. Is that easily debuggable? Do we even need to catch exceptions in tests if we don't expect the test to fail?  Even if the test is testing for something else?

Reply:
I'm not sure I understand the issue being raised here.

Helper functions can be introduced in the tests to produce valid input. All tests needing valid input can call these methods. If the valid input is no longer valid due to code changes, the specific function that tests for valid input will fail, and you can then fix the helper function to produce the valid input.

Reply:
There is no need to make the test results more inaccurate and errors untraceable by catching errors which are not supposed to be thrown. The errors that are not supposed to be thrown are OK being thrown in the tests if the developer breaks or changes the code. `Try-catch`ing in a test does not help as it would in an app where it helps for the developer to recover from the error or rather fail silently without producing a bad UX. In this case, I want the tests to fail fast in case there is anything wrong even in preparation of the tests

Reply:
Let's agree on a difference in opinion.